lib/sneakers/spawner.rb

```diff
@@ -14,7 +14,11 @@ def self.spawn
       worker_config = YAML.load(File.read(worker_group_config_file))
       worker_config.keys.each do |group_name|
         @pids << fork do
-          @exec_hash = {"WORKERS"=> worker_config[group_name]['classes'], "WORKER_COUNT" => worker_config[group_name]["workers"].to_s}
+          @exec_hash = {
+            "WORKERS"=> worker_config[group_name]['classes'],
+            "WORKER_COUNT" => worker_config[group_name]["workers"].to_s,
+            "PID_PATH" => worker_config[group_name]['pid_path'].to_s
```

Comment:
I'd say you're right, a series of `extract method` refactorings would go a long way towards making `Spawner.spawn` testable. 

As it stands, you'd have to stub out `Kernel.exec`, `Signal.trap`, and `Process.waitall` to be able to safely run this in a test - that's quite an interaction surface.